hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/net/NetworkTopology.java

```diff
@@ -196,10 +196,8 @@ protected Node getNodeForNetworkLocation(Node node) {
         loc = loc.substring(1);
       }
       InnerNode rack = (InnerNode) clusterMap.getLoc(loc);
-      if (rack == null) {
-        return null;
-      }
-      return new ArrayList<Node>(rack.getChildren());
+      return (rack == null) ? Collections.emptyList()
```

Comment:
Why can we use unmodifiable here?

Reply:
Ya, I guess we probably shouldn't.  My thinking wsa "

- Returning null is usually a bad idea, return Collections.emptyList()
- Collections.emptyList() is immutable
- To make the method consistent, the list of racks should be immutable too.

I will revert this to return an emptyList (new ArrayList<>(0))